remsfal-service/src/main/java/de/remsfal/service/boundary/project/ChatResource.java

```diff
@@ -0,0 +1,371 @@
+package de.remsfal.service.boundary.project;
+
+import de.remsfal.core.api.project.ChatEndpoint;
+import de.remsfal.core.json.project.ChatMessageJson;
+import de.remsfal.core.json.project.ChatSessionJson;
+import de.remsfal.core.model.project.ChatSessionModel;
+import de.remsfal.service.boundary.authentication.RemsfalPrincipal;
+import de.remsfal.service.control.ChatMessageController;
+import de.remsfal.service.control.ChatSessionController;
+import de.remsfal.service.entity.dto.ChatMessageEntity;
+import de.remsfal.service.entity.dto.ChatSessionEntity;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.*;
+import org.jboss.logging.Logger;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+@RequestScoped
+public class ChatResource extends ProjectSubResource implements ChatEndpoint {
+
+    @Inject
+    ChatSessionController chatSessionController;
+
+    @Inject
+    ChatMessageController chatMessageController;
+
+    @Inject
+    RemsfalPrincipal principal;
+
+
+    @Inject
+    Logger logger;
+
+
+    @Override
+    public Response createChatSession() {
+        String userId = principal.getId(); // Get user ID from session
+        if (userId == null || userId.isBlank()) {
```

Comment:
You do not need to check the authentication, because the authentication should be ensured by the PrincipalRequestFilter. Much more important is the authorization, because only owner and manager of a project should be able to create a new chat session. See checkPrivileges() in ProjectSubResource.

Reply:
I have a question regarding the authorization process. Shouldn't the chat function, *similar to a ticket*, be initiated by the client or tenant? Additionally, shouldn't the chat session, like a ticket, be managed and only closed or archived by the owner or manager? That was my initial understanding of a chat session—as something similar to a ticket